Algorithm.CSharp/BasicSetAccountCurrencyAlgorithm.cs

```diff
@@ -0,0 +1,92 @@
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Basic algorithm using SetAccountCurrency
+    /// </summary>
+    public class BasicSetAccountCurrencyAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private Symbol _btcEur;
+
+        /// <summary>
+        /// Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2018, 04, 04);  //Set Start Date
+            SetEndDate(2018, 04, 04);    //Set End Date
+            //Before setting any cash or adding a Security call SetAccountCurrency
+            SetAccountCurrency("EUR");
+            SetCash(100000);             //Set Strategy Cash
+
+            _btcEur = AddCrypto("BTCEUR").Symbol;
+        }
+
+        /// <summary>
+        /// OnData event is the primary entry point for your algorithm. Each new data point will be pumped in here.
+        /// </summary>
+        /// <param name="data">Slice object keyed by symbol containing the stock data</param>
+        public override void OnData(Slice data)
+        {
+            if (!Portfolio.Invested)
+            {
+                SetHoldings(_btcEur, 1);
+                Debug("Purchased Stock");
+            }
+        }
+
+        /// <summary>
+        /// This is used by the regression test system to indicate if the open source Lean repository has the required data to run this algorithm.
+        /// </summary>
+        public bool CanRunLocally { get; } = true;
+
+        /// <summary>
+        /// This is used by the regression test system to indicate which languages this algorithm is written in.
+        /// </summary>
+        public Language[] Languages { get; } = { Language.CSharp, Language.Python };
+
+        /// <summary>
+        /// This is used by the regression test system to indicate what the expected statistics are from running the algorithm
+        /// </summary>
+        public Dictionary<string, string> ExpectedStatistics => new Dictionary<string, string>
+        {
+            {"Total Trades", "1"},
+            {"Average Win", "0%"},
+            {"Average Loss", "0%"},
+            {"Compounding Annual Return", "-100.000%"},
+            {"Drawdown", "11.000%"},
+            {"Expectancy", "0"},
+            {"Net Profit", "-7.328%"},
+            {"Sharpe Ratio", "-12.15"},
+            {"Loss Rate", "0%"},
+            {"Win Rate", "0%"},
+            {"Profit-Loss Ratio", "0"},
+            {"Alpha", "-17.811"},
+            {"Beta", "1239.638"},
+            {"Annual Standard Deviation", "0.762"},
+            {"Annual Variance", "0.581"},
+            {"Information Ratio", "-12.169"},
+            {"Tracking Error", "0.761"},
+            {"Treynor Ratio", "-0.007"},
+            {"Total Fees", "$0.00"}
```

Comment:
Perhaps this algorithm could be updated to set the GDAX fee model so we can assert that fees are no longer debited in `USD` but in `EUR`.

Btw, I couldn't wait to try this with live GDAX with real EUR account and I can confirm it works as expected -- no more USD fees on EUR account 👍

Reply:
Sweet!
The `FeeModel` not in account currency case, is explicitly tested by `FeeModelNotUsingAccountCurrency` regression algorithm, using GDAX fee model 👍https://github.com/QuantConnect/Lean/blob/master/Algorithm.CSharp/FeeModelNotUsingAccountCurrency.cs